flang/lib/Optimizer/CodeGen/Target.cpp

```diff
@@ -810,6 +810,33 @@ struct TargetAArch64 : public GenericTarget<TargetAArch64> {
     return marshal;
   }
 
+  CodeGenSpecifics::Marshalling
+  integerArgumentType(mlir::Location loc,
```

Comment:
Does this work for both returns and arguments? If not how is the returns handled?

Reply:
Yes, returns are handled the same as arguments for non-pointer values.